packages/components/breadcrumbs/breadcrumbs.ts

```diff
@@ -0,0 +1,161 @@
+import {
+    AfterContentInit,
+    booleanAttribute,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChild,
+    ContentChildren,
+    DestroyRef,
+    Directive,
+    inject,
+    InjectionToken,
+    Input,
+    OnInit,
+    Provider,
+    QueryList,
+    TemplateRef,
+    ViewEncapsulation
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { RouterLink } from '@angular/router';
+import { KbqButton, KbqButtonModule, KbqButtonStyles } from '@koobiq/components/button';
+import { KbqComponentColors, KbqDefaultSizes } from '@koobiq/components/core';
+import { KbqDropdownModule } from '@koobiq/components/dropdown';
+import { KbqIconModule } from '@koobiq/components/icon';
+import { RdxRovingFocusGroupDirective, RdxRovingFocusItemDirective } from '@radix-ng/primitives/roving-focus';
+
+export interface KbqBreadcrumbsConfiguration {
+    max: number | null;
+    size: KbqDefaultSizes;
+    separator?: string;
+}
+
+export const KBQ_BREADCRUMBS_DEFAULT_CONFIGURATION: KbqBreadcrumbsConfiguration = {
+    max: 4,
+    size: 'normal'
+};
+
+/** Breadcrumbs options global configuration provider. */
+export const KBQ_BREADCRUMBS_CONFIGURATION = new InjectionToken<KbqBreadcrumbsConfiguration>(
+    'KBQ_BREADCRUMBS_CONFIGURATION',
+    { factory: () => KBQ_BREADCRUMBS_DEFAULT_CONFIGURATION }
+);
+
+/** Utility provider for `KBQ_BREADCRUMBS_CONFIGURATION`. */
+export const kbqBreadcrumbsConfigurationProvider = (configuration: KbqBreadcrumbsConfiguration): Provider => ({
+    provide: KBQ_BREADCRUMBS_CONFIGURATION,
+    useValue: configuration
+});
+
+@Directive({
+    selector: '[kbqBreadcrumbsSeparator]',
+    standalone: true
+})
+export class KbqBreadcrumbsSeparator {
+    readonly templateRef = inject(TemplateRef);
+}
+
+/**
+ * This directive is used to style breadcrumb buttons with default styling rules.
+ *
+ */
+@Directive({
+    standalone: true,
+    hostDirectives: [RdxRovingFocusItemDirective],
+    selector: '[kbq-button][kbqBreadcrumb]',
+    host: { class: 'kbq-breadcrumb' }
+})
+export class KbqDefaultBreadcrumbStyler implements OnInit {
+    button = inject(KbqButton, { optional: true, host: true });
+
+    ngOnInit() {
+        if (this.button) {
+            this.button.color = KbqComponentColors.Contrast;
+            this.button.kbqStyle = KbqButtonStyles.Transparent;
+        }
+    }
+}
+
+/**
+ * Directive provides a way to define a custom template for breadcrumb rendering, leveraging TemplateRef
+ */
+@Directive({
+    standalone: true,
+    selector: '[kbqBreadcrumbView]'
+})
+export class KbqBreadcrumbView {
+    templateRef = inject(TemplateRef);
+}
+
+/**
+ * Directive represents an individual breadcrumb item with optional support for router navigation and styling.
+ */
+@Directive({
+    standalone: true,
+    selector: 'kbq-breadcrumb-item',
+    host: {
+        class: 'kbq-breadcrumb'
+    }
+})
+export class KbqBreadcrumb {
```

Comment:
желательно чтобы селектор соответсвовал имени компонента
```suggestion
export class KbqBreadcrumbItem {
```

Reply:
реализовано